### PR TITLE
Initial Get-NewPackageFile and Get-RedirectedUri

### LINKED ISSUE #2: Create Get-NewPackageFile
Retrieves a file from either a web server or file share.

### PR DESCRIPTION
**Pull request by gjslavik**
Resolves #2. Initial creation of scripts.

Made minor tweaks to the Get-RedirectedUri script that Jeff wrote a while back. Just the order of comparing $null and some of the header info to match functionality correctly. The Get-NewPackageFile does not have any standard output when complete. Is this needed or desired? If so, what type of output should be generated?

Tested Get-RedirectedUri with multiple URIs, they all return as expected.
Tested Get-NewPackageFile with Source and Destination as file, UNC, and URI. Functionality worked as intended, or errors thrown as intended. Function copied files between local directories, from local to UNC, from UNC to UNC, from UNC to local, from URI to local, from URI to UNC, etc.

### CODE REVIEW

File: UMN-AutoPackager/public/Get-NewPackageFile.ps1 [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+.EXAMPLE
+Get-NewPackageFile -Source "https://webserver.xyz/Test/File1.txt" -Destination "\\fileserver\Temp\File1.txt"
+.INPUTS
+String
+.NOTES
+If the Destination parameter is only a file path and does not include a filename, the filename obtained from
+the Source parameter is used during the copy. Requires use of the "Get-RedirectedUri" function.
+#>
+    
+    [CmdletBinding(SupportsShouldProcess)]
```
Comment (thoen008) [inline UMN-AutoPackager/public/Get-NewPackageFile.ps1:35]: Any reason to not just use [CmdletBinding]()] which would include all the other parameters not just whatif?

Comment (thoen008) [reply UMN-AutoPackager/public/Get-NewPackageFile.ps1:35]: Reading up on it it looks like it doesn't enable those without adding that. Resolving the conversation.

File: UMN-AutoPackager/public/Get-RedirectedUri.ps1 [hunk 2]
```diff
@@ -0,0 +1,48 @@
+function Get-RedirectedUri {
```
Comment (jbolduan) [inline UMN-AutoPackager/public/Get-RedirectedUri.ps1:1]: This function needs to be moved to the /private directory.

File: UMN-AutoPackager/public/Get-NewPackageFile.ps1 [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+
+        [Parameter(Mandatory = $true,
+            HelpMessage = "Enter a valid Destination file path or UNC.")]
+        [ValidateNotNullOrEmpty()]
+        [string]$Destination
+    )
+
+    process {
+        try {
+            Write-Verbose "Verbose output enabled"
```
Comment (jbolduan) [inline UMN-AutoPackager/public/Get-NewPackageFile.ps1:50]: You should use the -Message "blah" on Write-Verbose calls to make things clear.  I'm only going to add this comment here, but applies to all instances of Write-Verbose.

I would also clean up the number of Write-Verbose instances you have.  There are way too many.  These execute every time the function is run but are outputted to null if the -Verbose switch isn't enabled.  This still takes up processing time.

**Review by thoen008 — APPROVED** [review]
Looks good to me.

**Review by jbolduan — CHANGES_REQUESTED** [review]
I see a couple changes that could/should be made.  Added comments.

File: UMN-AutoPackager/public/Get-RedirectedUri.ps1 (hunk 2 above)
Comment (gjslavik) [reply UMN-AutoPackager/public/Get-RedirectedUri.ps1:1]: Will be moved on next update. Resolving conversation.

File: UMN-AutoPackager/public/Get-NewPackageFile.ps1 (hunk 3 above)
Comment (gjslavik) [reply UMN-AutoPackager/public/Get-NewPackageFile.ps1:50]: Good info! Reducing and consolidating the number of Write-Verbose instances, and adding "-Message" to those that remain. Resolving coversation.

**Comment by gjslavik** [comment]
@jbolduan I moved Get-RedirectedUri to private per your request. Should Get-NewPackageFile move as well?

**Comment by gjslavik** [comment]
Added script functionality to properly handle "WhatIf" and "Confirm" along with "Verbose". Also added a return of "Get-Info" on the saved file so other scripts can work with output if needed.

File: UMN-AutoPackager/private/Get-RedirectedUri.ps1 [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+		Gets the real download URL from the redirection.
+	.DESCRIPTION
+		Used to get the real URL for downloading a file, this will not work if downloading the file directly.
+	.EXAMPLE
+		Get-RedirectedURI -URI "https://download.mozilla.org/?product=firefox-latest&os=win&lang=en-US"
+	.PARAMETER URL
+		URL for the redirected URL to be un-obfuscated
+	.NOTES
+		Code from: Redone per issue #2896 in core https://github.com/PowerShell/PowerShell/issues/2896
+	#>
```
Comment (jbolduan) [inline UMN-AutoPackager/private/Get-RedirectedUri.ps1:13]: We should standardize on help for functions being outside right above the function.

File: UMN-AutoPackager/public/Get-NewPackageFile.ps1 [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+    Get-NewPackageFile -Source "https://webserver.xyz/Test/File1.txt" -Destination "\\fileserver\Temp\File1.txt"
+    .INPUTS
+    String
+    .OUTPUTS
+    System.IO.FileInfo
+    Get-NewPackageFile returns information on the file saved in the location specified in the Destination parameter.
+    .NOTES
+    If the Destination parameter is only a file path and does not include a filename, the filename obtained from the Source parameter
+    is used during the copy. Requires use of the "Get-RedirectedUri" function.
+#>
```
Comment (jbolduan) [inline UMN-AutoPackager/public/Get-NewPackageFile.ps1:35]: See help comment on other function, we should standardize on this being above the function not inside.

**Review by jbolduan — CHANGES_REQUESTED** [review]
Only changes I think are worth making is moving the help outside the functions.  Otherwise it looks good.

**Review by thoen008 — APPROVED** [review]
I think it was decided to leave the comments under the function name. Otherwise I think everything looks good.

**Review by jbolduan — APPROVED** [review]
We agreed that comment help should be inside the functions based on community coding standards.

**Pull request merged**